static/app/views/replays/details.tsx

```diff
@@ -99,6 +104,26 @@ function ReplayDetails({
     );
   }
 
+  // If the user has navigated to the replay from an event, then we want to
+  // start the video at the time of the event.
+  if (!defined(initialTimeOffset) && defined(eventTimestamp) && startTimestampMs) {
+    // check if our event timestamp is within the range of the replay
+    const eventTimestampMs = new Date(eventTimestamp).getTime();
+    if (eventTimestampMs >= startTimestampMs) {
```

Comment:
The event timestamp that we have for the transactions are not precise, we don't have the ms, and there might be cases where we are going to have a time before our replay start time, in those cases we might want to just set our initial time offset to `0`